src/js/common/schemaform/FormApp.jsx

```diff
@@ -40,6 +40,16 @@ class FormApp extends React.Component {
     if (window.History) {
       window.History.scrollRestoration = 'manual';
     }
+
+    // If we start in the middle of a form, redirect to the beginning
+    const currentPath = this.props.location.pathname;
```

Comment:
As far as I can tell `this.props.location` == `this.props.currentLocation` == `this.props.router.location`. I'll use `currentLocation`, but we might consider revisiting this to see _why_ we've got three instances of the same object (basically).